### ISSUE TITLE
when cross-compiling it never uses pre-build (static) library and `zlib_installed` does not use the correct flags

### ISSUE THREAD

**Issue by BlackDex** [rust-lang/libz-sys#132]
Hello there,

I build a custom musl toolchain image with several libraries, to in the end build static Rust binaries.
I just recently found out that this crate doesn't use my pre-build static zlib package.

I noticed that it in the end will never use my pre-build binary, since I'm cross-compiling and the code shows it will always build no matter what. Bypassing this didn't resulted in a working version either, since the `zlib_installed` function does not use all the correct compiler flags.

To fix the flags was very easy by converting

```rust
let mut cmd = Command::new(compiler.path());
```
to
```rust
let mut cmd = compiler.to_command();
```

That solved that issue, and is probably what you want in the end.
I have created a PR for this #131 .

But, the part that it will never run that function on my custom build images, is a bit more difficult i think.

Now, I can move the `zlib_installed` check in front of the `build unconditionally` check, but that is probably not what you intended.
What would be a good way to still have pre-build libraries, and make sure the checks done for the unconditionally builds are being run when really needed?

**Comment by BlackDex** [comment]
I also just noticed, that it also will not mark it static, even when the correct environment variables are set.

**Comment by Byron** [comment]
I am too much out of my element here to chime in, but hope that the other maintainers can help figure out which path(s) to take here.

Generally I think it would be great to have a failing test that reproduces the particular issue(s) described here, so that a fix is protected from future regressions.

**Comment by BlackDex** [comment]
Anyone else who can provide some pointers?

**Comment by inferiorhumanorgans** [comment]
Well I'm attempting to cross compile for a target (dragonfly) that comes with zlib by default.  The main issue for me (and why I went looking for this issue in the first place) is that the bundled zlib doesn't compile (seemingly because it doesn't pull in stdio.h).

So for me, I'd prefer to see the cross compile case default to using the available libz with the option to override.

**Comment by wezm** [comment]
Can this issue be closed now that #131 is merged?

**Comment by Byron** [comment]
Thanks for the hint! Both seem so closely related that closing this issue pre-emptively seems acceptable. However, I invite those invested to validate the issue is actually fixed, and let us know here to reopen. Thank you.
